libflux/flux-core/src/formatter/tests.rs

```diff
@@ -72,6 +72,21 @@ fn funcs() {
     );
 }
 
+#[test]
+fn call_expr() {
+    // call function
+    assert_unchanged("a()");
+    assert_format("(a)()", "a()");
+    // call anonymous function
+    assert_unchanged("((a) => a)()");
+    // pipe anonymous function
+    assert_unchanged("(() => 1) |> f()");
+}
+#[test]
+fn function_expr() {
+    assert_unchanged("() => 1 and () => 2");
```

Comment:
Can you clarify what this line is testing? What would the failure state here look like?

Reply:
Would it just become `=> 1 and => 2` without the fix?

Reply:
A failure state would be to add parens around the righthand function literal. This shows that while precedence is lower than call expressions its higher than other expressions like `and` logical expressions.

Reply:
Technically this is not semantically valid syntax as you can't `and` two function literals, but we can still parse and format the syntax.